pkg/controller/machine/machine_controller_noderef.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package machine
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	apicorev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/klog"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha2"
+	capierrors "sigs.k8s.io/cluster-api/pkg/controller/error"
+	"sigs.k8s.io/cluster-api/pkg/controller/noderefutil"
+	"sigs.k8s.io/cluster-api/pkg/controller/remote"
+)
+
+var (
+	ErrNodeNotFound = errors.New("cannot find node with matching ProviderID")
+)
+
+func (r *ReconcileMachine) reconcileNodeRef(ctx context.Context, cluster *v1alpha2.Cluster, machine *v1alpha2.Machine) error {
+	// Check that the Machine hasn't been deleted or in the process.
+	if !machine.DeletionTimestamp.IsZero() {
+		return nil
+	}
+
+	// Check that the Machine doesn't already have a NodeRef.
+	if machine.Status.NodeRef != nil {
+		return nil
+	}
+
+	// Check that Cluster isn't nil.
+	if cluster == nil {
+		klog.V(2).Infof("Machine %q in namespace %q doesn't have a linked cluster, won't assign NodeRef", machine.Name, machine.Namespace)
```

Comment:
This was `klog.Warningf` before - you want to drop it to V2?

Reply:
Added the warning back, not sure if it actually gives any information to the user that they don't already know